c++/include/orc/Writer.hh

```diff
@@ -0,0 +1,294 @@
+     */
+    CompressionKind getCompression() const;
+
+    /**
+     * Set the encoding strategy.
+     */
+    WriterOptions& setEncodingStrategy(EncodingStrategy strategy);
+
+    /**
+     * Get the encoding strategy.
+     * @return if not set, return default value which is SPEED.
+     */
+    EncodingStrategy getEncodingStrategy() const;
+
+    /**
+     * Set the compression strategy.
+     */
+    WriterOptions& setCompressionStrategy(CompressionStrategy strategy);
+
+    /**
+     * Get the compression strategy.
+     * @return if not set, return default value which is speed.
+     */
+    CompressionStrategy getCompressionStrategy() const;
+
+    /**
+     * Set the writer version.
+     */
+    WriterOptions& setWriterVersion(WriterVersion version);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default value which is WriterVersion_ORIGINAL
+     */
+    WriterVersion getWriterVersion() const;
+
+    /**
+     * Set the padding tolerance.
+     */
+    WriterOptions& setPaddingTolerance(double tolerance);
+
+    /**
+     * Get the padding tolerance.
+     * @return if not set, return default value which is zero.
+     */
+    double getPaddingTolerance() const;
+
+    /**
+     * Set the memory pool.
+     */
+    WriterOptions& setMemoryPool(MemoryPool * memoryPool);
+
+    /**
+     * Get the strip size.
+     * @return if not set, return default memory pool.
+     */
+    MemoryPool * getMemoryPool() const;
+
+    /**
+     * Set the error stream.
+     */
+    WriterOptions& setErrorStream(std::ostream& errStream);
+
+    /**
+     * Get the error stream.
+     * @return if not set, return std::err.
+     */
+    std::ostream * getErrorStream() const;
+
+    /**
+     * Set whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     */
+    WriterOptions& setEnableStats(bool enable);
+
+    /**
+     * Get whether or not to write statistics (file statistics,
+     * stripe statistics, etc.)
+     * @return if not set, the default is true
+     */
+     bool getEnableStats() const;
+
+    /**
+     * Set whether or not to update string min/max stats
+     */
+    WriterOptions& setEnableStrStatsCmp(bool enable);
+
+    /**
+     * Get whether or not to update string min/max stats
+     * @return if not set, the default is false
+     */
+    bool getEnableStrStatsCmp() const;
+
+    /**
+     * Set whether or not to write row group index
+     */
+    WriterOptions& setEnableIndex(bool enable);
+
+    /**
+     * Get whether or not to write row group index
+     * @return if not set, the default is false
+     */
+    bool getEnableIndex() const;
+
+    /**
+     * Get writer timezone
+     * @return writer timezone
+     */
+    const Timezone* getTimezone() const;
+
+    /**
+     * Set writer timezone
+     * @param zone writer timezone name
+     */
+    WriterOptions& setTimezone(const std::string& zone);
```

Comment:
Let's not give this option. In the long term, it shouldn't matter and I'd rather not add it to the public api.